client/src/components/Production/ProductionModal.jsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Center,
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  Input,
+  ModalFooter,
+  Select,
+  Textarea,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+} from '@chakra-ui/react';
+import { TextField } from '@material-ui/core';
+import { Fragment, useEffect } from 'react';
+import useProductionModal from 'hooks/useProductionModal';
+import ProductionModalPart from 'components/Production/ProductionModalPart.jsx';
+import ProductionModalBike from 'components/Production/ProductionModalBike.jsx';
+
+const ProductionModal = () => {
+  const {
+    formStyle,
+    elementStyle,
+    isOpen,
+    onOpen,
+    onClose,
+    form,
+    colorRef,
+    quantityRef,
+    statusRef,
+    nameRef,
+    descriptionRef,
+    startDate,
+    endDate,
+    assemblyMachineRef,
+    noteRef,
+    profitMarginRef,
+    partRefs,
+    bikeRefs,
+    isEmptyField,
+    onClickPart,
+    onClickBike,
+    handleChange,
+    handleStartDateInput,
+    handleEndDateInput,
+    handleSubmit,
+  } = useProductionModal();
+
+  useEffect(() => {
+    handleChange();
+  }, [form, startDate, endDate, handleChange]);
+
+  return (
+    <>
+      <Center mt={4}>
+        <Button onClick={onOpen}>Produce</Button>
+      </Center>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Production</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <Button mr={3} onClick={onClickPart} colorScheme="teal">
+              Part
+            </Button>
+            <Button onClick={onClickBike} colorScheme="teal">
```

Comment:
```suggestion
            <Button onClick={onClickBike} colorScheme={form === 'Bike' ? "teal" : "gray"}>
```

Reply:
I don't agree with this. Both buttons will still be there, it's just one wouldn't be apparent? I did some changes that only one button would be there, take a look.

Reply:
@SiuYe Yup, your solution is good.

Can we change the button text to `Change production type`? It feels odd just have a button that says "part" or "bike".

You could use something like this maybe `onClick={form === 'Bike' ?  onClickPart : onClickBike}` since the button text won't be changing.

And drop the word `modal` from `Production modal for Bike`?

Reply:
done!